### PR TITLE
[Metal][ops] fix argmax/argmin for non-contiguous inputs (#160740)

### LINKED ISSUE #160740: MPS torch.argmax/torch.argmin fail for non-contiguous input
### 🐛 Describe the bug

As the title says:

```python
import torch

def test_argmax(x):
    device = x.device
    try:
        output = torch.argmax(x)
        print(f"  argmax test succeeds for device: {device}. output: {output}")
    except Exception as e:
        print(f"  argmax test fails for device: {device}: {e}")

x_cpu_contiguous = torch.randn(5, 5, device="cpu")
x_mps_contiguous = x_cpu_contiguous.to("mps")
x_cpu_transposed = x_cpu_contiguous.t()
x_mps_transposed = x_mps_contiguous.t()
x_cpu_strided = x_cpu_contiguous[::2, ::2]
x_mps_strided = x_mps_contiguous[::2, ::2]

print("contiguous tests:")
test_argmax(x_cpu_contiguous)
test_argmax(x_mps_contiguous)

print("transposed tests:")
test_argmax(x_cpu_transposed)
test_argmax(x_mps_transposed)

print("strided tests:")
test_argmax(x_cpu_strided)
test_argmax(x_mps_strided)
```

terminal output:
```
contiguous tests:
  argmax test succeeds for device: cpu. output: 8
  argmax test succeeds for device: mps:0. output: 8
transposed tests:
  argmax test succeeds for device: cpu. output: 16
  argmax test fails for device: mps:0: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.
strided tests:
  argmax test succeeds for device: cpu. output: 0
  argmax test fails for device: mps:0: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.
```

### Versions

nightly

### PR DESCRIPTION
**Pull request by versi379**
## Problem

`torch.argmax` / `torch.argmin` on MPS fail on non-contiguous inputs (transposed, strided slices) with: `view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.` CPU works fine on the same inputs.

The no-dim path in `argmax_argmin_out_mps` flattens the input via `apparent_in_shape = [num_elements]`, and the `Placeholder` constructor reshapes via `.view()`, which requires contiguous memory.

## Reproducer (from the issue)

```python
import torch
x = torch.randn(5, 5, device="mps")
torch.argmax(x.t())          # transposed, raises
torch.argmax(x[::2, ::2])    # strided, raises
```

## How

In `argmax_argmin_out_mps` (`aten/src/ATen/native/mps/operations/ReduceOps.mm`), call `.contiguous()` on the input at the top of the function. No-op for already-contiguous tensors; for non-contiguous inputs it materializes a contiguous copy so the downstream `Placeholder` / `.view()` succeeds. Sibling fix to #187039 (argmax/argmin NaN handling) in the same file.

## Test

```
python test/test_mps_issue_160740.py -v
```

Cases: transposed 2D, strided 2D slice, with-dim on transposed, large column-strided slice, 3D permuted. All match CPU output on Apple M4 Pro.

Happy to relocate the test into `test/test_mps.py` if preferred.

Fixes #160740.

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/ReduceOps.mm [hunk 1]
```diff
@@ -796,22 +796,23 @@ static void argmax_argmin_out_mps(const Tensor& input_t,
                                   const std::string& func_name) {
   using CachedGraph = MPSUnaryCachedGraph;
 
+  // Contiguous input required: the no-dim path reshapes to 1-D via
```
Comment (Skylion007) [inline aten/src/ATen/native/mps/operations/ReduceOps.mm:721]: Huh, is there no a way to pass the striding instead?

Comment (versi379) [reply aten/src/ATen/native/mps/operations/ReduceOps.mm:721]: Thanks for the feedback!

Right. The `.contiguous()` (and thus the copy) is only needed when no dim is passed in argmax/argmin (flatten path). On macOS 15+ Placeholder passes strides through to MPSGraph for the dim-provided case, so we can skip it there.

Maybe we can differentiate the two cases:

```cpp
auto input = dim.has_value() ? input_t : input_t.contiguous();
```

**Review by malfet — CHANGES_REQUESTED** [review]
Let me just migrate argmax/argmin to Metal, as max/min has been migrated recently

File: aten/src/ATen/native/mps/operations/ReduceOps.mm (hunk 1 above)
Comment (versi379) [reply aten/src/ATen/native/mps/operations/ReduceOps.mm:721]: Quick repro:

```python
  import torch
  x = torch.tensor([[1, 2, 3], [4, 5, 6]], dtype=torch.float, device="mps")
  xt = x.t()
  print(f"xt:           {xt.cpu().tolist()}")
  print(f"  shape={tuple(xt.shape)}, strides={xt.stride()}, contiguous={xt.is_contiguous()}")
  print(f"argmax(dim=0): {xt.argmax(dim=0).cpu().tolist()}")  # Case A: strided path, no copy
  print(f"argmax():      {xt.argmax().item()}")               # Case B: flatten, semantic copy
  ```

  Output:
  ```
xt:            [[1.0, 4.0], [2.0, 5.0], [3.0, 6.0]]
shape=(3, 2), strides=(1, 3), contiguous=False
argmax(dim=0): [2, 2]
argmax():      5
  ```

**Comment by kurtamohler** [comment]
Will be fixed by #187304
